### PR TITLE
 [Tests] Fix the make rule that generate the test projects. Fixes #5638

### LINKED ISSUE #5638: generate-test-projects target in BCLTests fails to compile
### Steps to Reproduce

1. run `make generate-test-projects` in tests/bcl-test/BCLTests
2. it fails to compile

### Expected Behavior

No compile error.

### Actual Behavior

```
Build FAILED.

"/Users/alexander/dev/xamarin-macios/xamarin-macios/tools/bcl-test-importer/BCLTestImporter/BCLTestImporter.sln" (default target) (1) ->
"/Users/alexander/dev/xamarin-macios/xamarin-macios/tools/bcl-test-importer/BCLTestImporter/BCLTestImporter.csproj" (default target) (2) ->
(CoreCompile target) ->
  Program.cs(168,52): error CS0117: 'BCLTestProjectGenerator' does not contain a definition for 'Generate' [/Users/alexander/dev/xamarin-macios/xamarin-macios/tools/bcl-test-importer/BCLTestImporter/BCLTestImporter.csproj]
  Program.cs(186,47): error CS0117: 'RegisterTypeGenerator' does not contain a definition for 'GenerateCode' [/Users/alexander/dev/xamarin-macios/xamarin-macios/tools/bcl-test-importer/BCLTestImporter/BCLTestImporter.csproj]

    0 Warning(s)
    2 Error(s)
```

### PR DESCRIPTION
**Pull request by mandel-macaque**
Fixed the rule to generate the tests. This branch adds missing test dlls that the make rule check, or it will give an error stating that tests are missing.

This feature should be added to xharness and an issue was created for that:  https://github.com/xamarin/maccore/issues/1447

Fixes https://github.com/xamarin/xamarin-macios/issues/5638

### CODE REVIEW

File: tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs [hunk 1]
```diff
@@ -853,8 +862,10 @@ public bool AllTestAssembliesAreRan (out Dictionary<Platform, List<string>> miss
 		{
 			missingAssemblies = new Dictionary<Platform, List<string>> ();
 			foreach (var platform in new [] {Platform.iOS, Platform.TvOS}) {
-				var testDir = wasDownloaded ? BCLTestAssemblyDefinition.GetTestDirectoryFromMonoPath (MonoRootPath, platform)
-					: BCLTestAssemblyDefinition.GetTestDirectoryFromDownloadsPath (GetReleaseDownload (platform), platform); 
+				Console.WriteLine ($"Was downloaded? {wasDownloaded}");
```
Comment (rolfbjarne) [inline tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs]: CWL

File: tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs [hunk 2]
```diff
@@ -853,8 +862,10 @@ public bool AllTestAssembliesAreRan (out Dictionary<Platform, List<string>> miss
...
+				var testDir = wasDownloaded ? BCLTestAssemblyDefinition.GetTestDirectoryFromDownloadsPath (GetReleaseDownload (platform), platform)
+					: BCLTestAssemblyDefinition.GetTestDirectoryFromMonoPath (MonoRootPath, platform);
+				Console.WriteLine ($"Test direcotry is {testDir}");
```
Comment (rolfbjarne) [inline tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs]: CWL

Comment (mandel-macaque) [reply tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs]: Gosh, I was debugging that haha, will remove, thx!

**Comment by mandel-macaque** [comment]
build

**Comment by mandel-macaque** [comment]
build

**Comment by mandel-macaque** [comment]
Merging because the failures are known:

* WatchOS crashes: https://github.com/xamarin/maccore/issues/581
* monotouch failure: https://github.com/xamarin/maccore/issues/1453

**Pull request merged**